### PR TITLE
New subprojects

### PR DESCRIPTION
**Pull request by nhibberd**
- Created empty core projects for autoscaling, rds and ec2 (will be used to help upstream dependencies)
- split cli executables to a separate sub project (`mismi-cli`)
- created `mismi-openssl`

#183 

@markhibberd @jystic

### CODE REVIEW

File: mismi-autoscaling-core/mismi-autoscaling-core.cabal [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+synopsis:              AWS library
+category:              AWS
+cabal-version:         >= 1.8
+build-type:            Simple
+description:           mismi-autoscaling-core.
+
+library
+  build-depends:
+                       base                            >= 3          && < 6
+                     , ambiata-p
```
Comment (erikd-ambiata) [inline mismi-autoscaling-core/mismi-autoscaling-core.cabal]: Any reason for the `6` upper bound on base? Ghc 8.0 shipped with `4.9` and 8.2 is likely to ship with `4.10`.

File: mismi-ec2-core/mismi-ec2-core.cabal [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+synopsis:              AWS library
+category:              AWS
+cabal-version:         >= 1.8
+build-type:            Simple
+description:           mismi-ec2-core.
+
+library
+  build-depends:
+                       base                            >= 3          && < 6
+                     , ambiata-p
```
Comment (erikd-ambiata) [inline mismi-ec2-core/mismi-ec2-core.cabal]: Again, 6?

**Comment by erikd-ambiata** [comment]
I suppose the upper bound of `6` on base is not that big an issue, just thought it was a little strange.

:hamburger:

**Comment by jystic** [comment]
👍 but, as discussed IRL, another PR that makes it so you don't need call `withOpenSSL` would be great

**Comment by nhibberd** [comment]
@erikd-ambiata suggesting it should be 5 or? Was just keeping inline with other modules in mismi

**Comment by jystic** [comment]
@nhibberd I'd say it should be 5 if you feel like changing it, not a big deal though

**Comment by markhibberd** [comment]
👍

**Comment by erikd-ambiata** [comment]
@nhibberd Yeah, from what I've seen, 5 is what most of the external Haskell community seems to be using.

**Comment by nhibberd** [comment]
Have changed base dep and added an sqs-core

**Pull request merged**